### PR TITLE
add a Source::parse arm for sparse repos (keeping it in the scheme). …

### LINKED ISSUE #2938: Source::parse doesn't recognize sparse scheme?
Our github and local crate repository mirrors are using the sparse scheme, but I suspect that `copy_project_with_explicit_deps_on_all_transitive_proc_macros` is failing on this inside `Source::parse`:

```
Caused by:
    0: Failed to copy project with proc macro deps made direct
    1: Couldn't parse source "sparse+https://artifactory.london.quadraturecapital.com/artifactory/api/cargo/cargo-remote/index/": Didn't recognise scheme
```
Am I missing something about how this is supposed to work?

### PR DESCRIPTION
**Pull request by sthornington**
Proc macro resolution wasn't working properly for sparse repos.  This fixes #2938.

### CODE REVIEW

**Comment by sthornington** [comment]
I wrote a test for it, but was unable to run it - cargo test seemed to want some runfiles stuff and looking at that in crates.io lead me to believe that these things are not able to be tested from within cargo.

File: crate_universe/src/metadata/cargo_tree_resolver.rs [hunk 1]
```diff
@@ -936,6 +941,25 @@ mod test {
...
 
+    #[test]
+    fn parse_sparse_source() {
+        let source = Source::parse(
+            "sparse+https://github.com/rust-lang/crates.io-index",
+            "1.0.1".to_owned(),
+        )
+        .unwrap();
+        // sparse we want to leave the augmented scheme in there.
+        // cargo_toml::RegistryDetail::registry_index explicitly supports
```
Comment (illicitonion) [inline crate_universe/src/metadata/cargo_tree_resolver.rs:952]:
```suggestion
        // cargo_toml::DependencyDetail::registry_index explicitly supports
```

**Review by illicitonion — APPROVED** [review]
LGTM, thanks so much!

**Comment by sthornington** [comment]
Thanks, let me know if there is anything else for me to do to get this merged!

**Comment by sthornington** [comment]
Thanks everyone for getting this in.  Can I ask for a point release which includes this, or when is 0.53.1 otherwise planned?  Or can I do start it myself?  Thanks again.

**Pull request merged**